addons/link_tracker/models/link_tracker.py

```diff
@@ -112,22 +110,10 @@ def _compute_redirected_url(self):
     @api.model
     @api.depends('url')
     def _get_title_from_url(self, url):
-        try:
-            head = requests.head(url, allow_redirects=True, timeout=5)
-            if (
-                    int(head.headers.get('Content-Length', 0)) > URL_MAX_SIZE
-                    or
-                    'text/html' not in head.headers.get('Content-Type', 'text/html')
-            ):
-                return url
-            # HTML parser can work with a part of page, so ask server to limit downloading to 50 KB
-            page = requests.get(url, timeout=5, headers={"range": "bytes=0-50000"})
-            p = html.fromstring(page.text.encode('utf-8'), parser=html.HTMLParser(encoding='utf-8'))
-            title = p.find('.//title').text
-        except:
-            title = url
-
-        return title
+        link_preview = self.env['mail.link.preview']._get_link_preview_from_url(url, requests.Session())
```

Comment:
Continuing proposal for using a tool: it would be "link_preview = mail_tools.get_link_preview_from_url(url, ...)", which reads better to me than having to use a model